Added a function to PyModule to load a module from a string of Python

kmp1 opened the pull request:
I have a situation where I have a blob of Python code that I want to load into a module so that I can later pluck functions out of it to call throughout my application, import it, etc, etc.  

As such I figured it made sense to add the function defined in this pull request to the `PyModule` struct so you can construct such a struct containing some code as a kind of sibling function to the `new` and `import` functions you already have.

Is this something you would like?  If not, can you explain how I would get the same functionality from outside of the library?

konstin commented:
This look like a nice feature! Could you add a usage example as unit test? `tests/test_module` should be a good location.

The ffi calls you made only work the python 3 (see the [travis log](https://travis-ci.org/PyO3/pyo3/jobs/380644285) for more information), so you likely want to add a `#[cfg(Py_3)]` to the function.

kmp1 commented:
Thanks for the feedback!  

I added `#[cfg(Py_3)]` to the function and replaced two `as_ptr` calls with one that I noticed and thought it would be better to do it only once for clarity.

And I added a test to `test/test_module.rs` - I hope that's fine. Cargo test looks like this now...

```bash
     Running target\debug\deps\test_module-7dcfb66d3201b564.exe

running 2 tests
test test_module_from_code ... ok
test test_module_with_functions ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

konstin commented:
The test case looks good :+1:

For some reason I don't understand, the build is still failing on travis ci for python. This is weird since it's working perfectly for me locally and the trait in question (`ToPyObject`) is definitely defined in the prelude. I'm unfortunately not sure what to do about that. Could you try rebasing onto or merging with master so that we get some more information in the travis log?

konstin commented:
Ah well, I found the culprit: Travis merges the pr with master before building. On master, there are specific imports instead of a prelude import. So rebasing onto master and changing

```
use pyo3::{PyDict, PyModule, PyObject, PyResult, Python};
```

into 

```
use pyo3::{PyDict, PyModule, PyObject, PyResult, Python, ToPyObject};
```

should make travis pass, and then I can finally merge this.

kmp1 commented:
oh, did I do something awful here with the git??  I was trying to rebase my fork with your master and it seems like I collected up some commits I should not have.  Should I start this again so you get a cleaner PR?

kmp1 commented:
Oh, that's amazing, thank you!  :-)  Now there's 3 commits which should match up nicely with this conversation - I hope it's all good now.

konstin commented:
Perfect, thank you!
